File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-3.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int GoodNodes(TreeNode root) {
        return DFS(root, root.val);
    }

    private int DFS(TreeNode node, int maxValue)
    {
        if (node is null) return 0;

        var result = node.val >= maxValue ? 1 : 0;
        var max = Math.Max(node.val, maxValue);

        result += DFS(node.left, max);
        result += DFS(node.right, max);
        
        return result;
    }
}

File: Data Structures & Algorithms/sqrtx/submission-3.py
import math

class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0: return 0
        if x == 1: return 1

        for left in range(1, x):
            y = left * left
            if y == x:
                return left
            elif y >= x:
                return left-1